thorlcr/activities/diskread/thdiskreadslave.cpp

```diff
@@ -68,14 +68,24 @@ class CDiskReadSlaveActivityRecord : public CDiskReadSlaveActivityBase, implemen
     // return a ITranslator based on published format in part and expected/format
     ITranslator *getTranslators(IPartDescriptor &partDesc)
     {
-        unsigned projectedFormatCrc = helper->getProjectedFormatCrc();
-        IOutputMetaData *projectedFormat = helper->queryProjectedDiskRecordSize();
-        IPropertyTree const &props = partDesc.queryOwner().queryProperties();
-        Owned<IOutputMetaData> publishedFormat = getDaliLayoutInfo(props);
-        unsigned publishedFormatCrc = (unsigned)props.getPropInt("@formatCrc", 0);
         RecordTranslationMode translationMode = getTranslationMode(*this);
+
         unsigned expectedFormatCrc = helper->getDiskFormatCrc();
         IOutputMetaData *expectedFormat = helper->queryDiskRecordSize();
+
+        Linked<IOutputMetaData> publishedFormat;
+        unsigned publishedFormatCrc = 0;
+
+        const char *kind = queryFileKind(&partDesc.queryOwner());
+        if (strisame(kind, "flat") || (RecordTranslationMode::AlwaysDisk == translationMode))
+        {
+            IPropertyTree const &props = partDesc.queryOwner().queryProperties();
+            publishedFormat.setown(getDaliLayoutInfo(props));
+            publishedFormatCrc = (unsigned)props.getPropInt("@formatCrc", 0);
+        }
```

Comment:
If this isn't true should published be set equal to expected ?

Reply:
no, it's correct to be 0, if missing.
It won't make any difference, but should not set even if @formatCrc is present if 'publishedFormat' is not set. Will change.